cmake/Hipify.cmake

```diff
@@ -79,8 +79,7 @@ function(hipify)
   if(HIPIFY_CONFIG_FILE)
     set(HIPIFY_COMMAND
       ${HIPIFY_DIR}/hipify_cli.py
-      --config-file ${HIPIFY_CONFIG_FILE}
-      --dump-dict-file ${HIPIFY_DICT_FILE}
```

Comment:
set a default value for dump-dict-file in json file scenario.

Reply:
@pruthvistony This is tricky because the default value we want for dump-dict-file in the CMake hipify case seems to use a CMake variable: `set(HIPIFY_DICT_FILE ${CMAKE_BINARY_DIR}/hipify_output_dict_dump.txt)`. That means we can't have the default value in the json file exactly match the above CMake variable. Thoughts?

Reply:
There are few points to take care and update the design accordingly.
1) dumping of hipify dict is a must to support get_hipified_list() API, which is an external API, which can be called from user after the hipify_cli.py trigger either through config file or specifying parameters.
2) We shouldnt control the dict dumping ON/OFF through json-config parameter [code-link](https://github.com/ROCmSoftwarePlatform/hipify_torch/pull/14/files/d20f2d94505a2124fbb7ac380ccfcb2b4a995754#diff-0b7bddf11afd4d0bda06f4377234786c0d9354f2c038e8dbd828dbba1837bcdaR94). So if we want configurability we can control only the name and have default value for 'HIPIFY_DICT_FILE'.

let me know your thoughts.
cc @jithunnair-amd @bmedishe

Reply:
For the scope of this PR, I'm okay with having `dump-dict-file` as a separate argument to hipify_cli.py. But we should try to see if there's a way to unify all hipify_cli.py arguments into the json file while satisfying the requirements of the CMake hipify flow.